File: lab1/popov-kirill/ex3(studio)/ex3(studio)/Tags.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ex3_studio_
{
    class Tags
    {
        private string[] mdTags = { @"\s\>", @"\#\#\#\#\#\#\s", @"\#\#\#\#\#\s", @"\#\#\#\#\s", @"\#\#\#\s", @"\#\#\s", @"\#\s", @"\*",
            @"\~\~", @"\-\s", @"\`" };
        private string[] htmlTags = { "blockquote", "h6", "h5", "h4", "h3", "h2", "h1", "i", "s", "li", "code" };
        private string mainPattern = @"+[\w+\s+\'+\!+\?+\.+\:+]+";
        private bool[] isSingle = { true, true, true, true, true, true, true, false, false, true, false };
        public int count;
        public Tags()
        {
            count = mdTags.Length;
        }

        public int Count { get { return count; } }

        public string Pattern(int iterator)
        {
            if (isSingle[iterator])
            {
                return mdTags[iterator] + mainPattern + @"<br>";
            }
            else
            {
                return mdTags[iterator] + mainPattern + mdTags[iterator];
            }
        }

        public string MDTag(int iterator)
        {
            return mdTags[iterator];
        }

        public string HTMLTag(int iterator)
        {
            return htmlTags[iterator];
        }
    }
}

File: lab1/task3/tags.py
import re


def edit_tags(text: str) -> str:
    text = re.sub('\n', '<br>', text)

    text = re.sub(r'###### (.+?)<br>', r'<h6>\1</h6> <br>', text)
    text = re.sub(r'##### (.+?)<br>', r'<h5>\1</h5> <br>', text)
    text = re.sub(r'#### (.+?)<br>', r'<h4>\1</h4> <br>', text)
    text = re.sub(r'### (.+?)<br>', r'<h3>\1</h3> <br>', text)
    text = re.sub(r'## (.+?)<br>', r'<h2>\1</h2> <br>', text)
    text = re.sub(r'# (.+?)<br>', r'<h1>\1</h1> <br>', text)

    text = re.sub(r'(~~)(.+?)\1', r'<strike>\2</strike>', text)

    text = re.sub(r'(```)(.+?)\1', r'<code>\2</code>', text)

    text = re.sub(r'\* (\w+?)((<br>)|\Z)', r'<li>\1</li>', text)

    text = re.sub(r'((\*\*)|(__))(.+?)\1', r'<strong>\4</strong>', text)

    text = re.sub(r'(\*|_)(.+?)\1', r'<em>\2</em>', text)

    text = re.sub(r'<br>>(.+?)<br>', r'<br> <blockquote>\1</blockquote> <br>', text)

    return text
